app/commands/source/add_value_command.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+module Source
+  class AddValueCommand < ApplicationCommand
+    usage do
+      no_command
+    end
+
+    keyword :name do
+      desc "Name for the new value"
+      required
+    end
+
+    keyword :attribute_friendly_id do
+      desc "Friendly ID of the primary attribute to add the value to"
+      required
+    end
+
+    def execute
+      frame("Adding new value") do
+        ActiveRecord::Base.transaction do
+          setup!
+          create_value!
+          create_attributes_value!
+        rescue ActiveRecord::Rollback => e
```

Comment:
I think its easier to read keeping the exits in method? Saves having to raise simply to catch and exit?

Reply:
Since there are multiple db inserts to multiple tables, I opted to do it as a transaction and raising a rollback seems to be the common practice for cancelling/failing a transaction. Also it DRYs up the error handling a bit rather than putting `exit(1)` in 4 spots

Reply:
But instead you have a raise in 4 spots? Is that actually dryer?

Reply:
I think the fact this relies on `raise ActiveRecord::Rollback` is what makes this not sit well with me. We're not consolidating, we're adding a layer of redirect + requiring our top-level function to have a wrapped exception handler?

If we decide this pattern is preferred, all commands should be updated to match as well. The goal here is a consistent system. Happy to hash this out IRL to find the right _next best_ pattern

Reply:
By moving the creation logic out of the script and into the Value model, I was able to remove most of these exception handling, so I switched it back to the log+exit 😄